keadm/app/cmd/util/ubuntuinstaller.go

```diff
@@ -433,22 +435,68 @@ func (u *UbuntuOS) InstallKubeEdge() error {
 	//Currently it is missing and once checksum is in place, checksum check required
 	//to be added here.
 	filename := fmt.Sprintf("kubeedge-v%s-linux-%s.tar.gz", u.KubeEdgeVersion, arch)
+	checksumFilename := fmt.Sprintf("checksum_kubeedge-v%s-linux-%s.txt", u.KubeEdgeVersion, arch)
 	filePath := fmt.Sprintf("%s%s", KubeEdgePath, filename)
 	fileStat, err := os.Stat(filePath)
 	if err == nil && fileStat.Name() != "" {
 		fmt.Println("Expected or Default KubeEdge version", u.KubeEdgeVersion, "is already downloaded")
 		goto SKIPDOWNLOADAND
 	}
 
-	//Download the tar from repo
-	dwnldURL = fmt.Sprintf("cd %s && wget -k --no-check-certificate --progress=bar:force %s/v%s/%s", KubeEdgePath, KubeEdgeDownloadURL, u.KubeEdgeVersion, filename)
-	cmd = &Command{Cmd: exec.Command("sh", "-c", dwnldURL)}
-	err = cmd.ExecuteCmdShowOutput()
-	errout = cmd.GetStdErr()
-	if err != nil || errout != "" {
-		return fmt.Errorf("%s", errout)
+	for i := 0; i < downloadRetryTimes; i++ {
+		//Download the tar from repo
+		dwnldURL = fmt.Sprintf("cd %s && wget -k --no-check-certificate --progress=bar:force %s/v%s/%s", KubeEdgePath, KubeEdgeDownloadURL, u.KubeEdgeVersion, filename)
+		cmd = &Command{Cmd: exec.Command("sh", "-c", dwnldURL)}
+		err = cmd.ExecuteCmdShowOutput()
+		errout = cmd.GetStdErr()
+		if err != nil || errout != "" {
+			return fmt.Errorf("%s", errout)
+		}
+		fmt.Println()
+
+		//Download the checksum file from repo
+		dwnldURL = fmt.Sprintf("cd %s && wget -k --no-check-certificate --progress=bar:force %s/v%s/%s", KubeEdgePath, KubeEdgeDownloadURL, u.KubeEdgeVersion, checksumFilename)
+		cmd = &Command{Cmd: exec.Command("sh", "-c", dwnldURL)}
+		err = cmd.ExecuteCmdShowOutput()
+		errout = cmd.GetStdErr()
+		if err != nil || errout != "" {
+			return fmt.Errorf("%s", errout)
+		}
+		fmt.Println()
+
+		//Verify the tar with checksum
+		fmt.Printf("%s checksum: \n", filename)
+		cmdStr := fmt.Sprintf("cd %s && sha512sum %s | awk '{split($0,a,\"[ ]\"); print a[1]}'", KubeEdgePath, filename)
+		cmd = &Command{Cmd: exec.Command("sh", "-c", cmdStr)}
+		cmd.ExecuteCommand()
+		checksum1 := cmd.GetStdOutput()
+		fmt.Printf("%s \n\n", cmd.GetStdOutput())
+
+		fmt.Printf("%s content: \n", checksumFilename)
+		cmdStr = fmt.Sprintf("cd %s && cat %s", KubeEdgePath, checksumFilename)
+		cmd = &Command{Cmd: exec.Command("sh", "-c", cmdStr)}
+		cmd.ExecuteCommand()
+		checksum2 := cmd.GetStdOutput()
+		fmt.Printf("%s \n\n", cmd.GetStdOutput())
+
+		if checksum1 == checksum2 {
```

Comment:
can we rename checksum1 and checksum2 to desiredChecksum and actualChecksum ?

Reply:
@sids-b Thanks for your comment. The code has been updated accordingly.